packages/opentelemetry-types/src/trace/span.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SpanContext } from './span_context';
+import { Status } from './status';
+
+/**
+ * An interface that represents a span. A span represents a single operation
+ * within a trace. Examples of span might include remote procedure calls or a
+ * in-process function calls to sub-components. A Trace has a single, top-level
+ * "root" Span that in turn may have zero or more child Spans, which in turn
+ * may have children.
+ */
+export interface Span {
+  /**
+   * Returns the SpanContext object associated with this Span.
+   *
+   * @returns the SpanContext object associated with this Span.
+   */
+  context(): SpanContext;
+
+  // /**
+  //  * # TODO
+  //  * Returns the Tracer object used to create this Span.
+  //  * https://github.com/open-telemetry/opentelemetry-specification/issues/21
+  //  */
+  // tracer(): Tracer;
+
+  /**
+   * Sets an attribute to the span.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute. If the value is a typeof object
```

Comment:
This would be problematic for Datadog since we currently accept values that are not JSON serializable and we have a custom serializer that knows how to handle them.